ibis/common/typing.py

```diff
@@ -2,58 +2,182 @@
 
 import sys
 from typing import (
-    TYPE_CHECKING,
     Any,
-    ForwardRef,
-    Iterable,
-    Mapping,
-    Tuple,
+    Dict,
+    Generic,  # noqa: F401
+    Optional,
     TypeVar,
-    Union,
+    get_args,
+    get_origin,
 )
 
-import toolz
-
-# TODO(kszucs): try to use inspect.get_annotations() backport instead
-
-if sys.version_info >= (3, 9):
-
-    @toolz.memoize
-    def evaluate_typehint(hint, module_name=None) -> Any:
-        if isinstance(hint, str):
-            hint = ForwardRef(hint)
-        if isinstance(hint, ForwardRef):
-            if module_name is None:
-                globalns = {}
-            else:
-                globalns = sys.modules[module_name].__dict__
-            return hint._evaluate(globalns, locals(), frozenset())
-        else:
-            return hint
-
-else:
-
-    @toolz.memoize
-    def evaluate_typehint(hint, module_name) -> Any:
-        if isinstance(hint, str):
-            hint = ForwardRef(hint)
-        if isinstance(hint, ForwardRef):
-            if module_name is None:
-                globalns = {}
-            else:
-                globalns = sys.modules[module_name].__dict__
-            return hint._evaluate(globalns, locals())
-        else:
-            return hint
-
-
-if TYPE_CHECKING:
-    import ibis.expr.datatypes as dt
-    import ibis.expr.schema as sch
-
-    SupportsSchema = TypeVar(
-        "SupportsSchema",
-        Iterable[Tuple[str, Union[str, dt.DataType]]],
-        Mapping[str, Union[str, dt.DataType]],
-        sch.Schema,
-    )
+from typing_extensions import get_type_hints as _get_type_hints
+
+from ibis.common.caching import memoize
+
+Namespace = Dict[str, Any]
+
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+@memoize
+def get_type_hints(
+    obj: Any,
+    include_extras: bool = True,
+    include_properties: bool = False,
+) -> dict[str, Any]:
+    """Get type hints for a callable or class.
+
+    Extension of typing.get_type_hints that supports getting type hints for
+    class properties.
+
+    Parameters
+    ----------
+    obj : Any
+        Callable or class to get type hints for.
+    include_extras : bool, default True
```

Comment:
```suggestion
    include_extras
```

Likewise the default is visible in the function signature, no need to indicate it here.